client/lib/domains/utils.js

```diff
@@ -46,7 +46,7 @@ function getDomainNameFromReceiptOrCart( receipt, cart ) {
 		domainRegistration = cartItems.getDomainRegistrations( cart )[ 0 ];
 	}
 
-	if ( domainRegistration ) {
+	if ( domainRegistration && isEmpty( receipt.failed_purchases ) ) {
```

Comment:
`receipt` may be empty in this context. That's why there is check for it truthiness in line 41:
```js
if ( receipt && ! isEmpty( receipt.purchases ) ) {
```

I think we can leave early this function if this condition is true:
```js
if ( receipt && ! isEmpty( receipt. failed_purchases ) ) {
    return null;
}
```